### PR TITLE
1.99 third time lucky

### PR DESCRIPTION
**Pull request by katSchmid**

### CODE REVIEW

**Comment by chris-cooper** [comment]
Could you please change `main` to `upstream-1.99` (a branch I just pushed) to give a clean diff?

image

File: ThirdParty.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
     "license": [
       "MIT"
     ],
-    "version": "3.16.2",
+    "version": "3.15.0",
```
Comment (chris-cooper) [inline ThirdParty.json]: Are these versions going backwards expected?

File: package.json [hunk 2]
```diff
@@ -73,7 +73,7 @@
   },
```
Comment (chris-cooper) [inline package.json:74]: The package name will have to change for the NPM publish to work correctly from codeship

https://github.com/PropellerAero/cesium/blob/main/package.json#L2

File: package.json [hunk 3]
```diff
@@ -73,7 +73,7 @@
...
   "devDependencies": {
     "@rollup/plugin-commonjs": "^22.0.2",
```
Comment (chris-cooper) [inline package.json:76]: I notice we're doing something special with CameraFlightPath
https://github.com/PropellerAero/cesium/blob/main/package.json#L41

File: package.json [hunk 4]
```diff
@@ -73,7 +73,7 @@
...
-    "@rollup/plugin-node-resolve": "^14.0.0",
+    "@rollup/plugin-node-resolve": "^15.0.1",
     "aws-sdk": "^2.1210.0",
     "chokidar": "^3.5.3",
```
Comment (chris-cooper) [inline package.json:79]: Do you want to include the standard codeship scripts?

https://github.com/PropellerAero/cesium/blob/main/package.json#L126-L127

**Review by chris-cooper — COMMENTED** [review]
Is `Gltf3DTileContent` etc no longer needed? https://github.com/PropellerAero/cesium/commit/ab5e8839f7d510e8f23b30eee3fc7ec59e9984c3

File: package.json (hunk 2 above)
Comment (katSchmid) [reply package.json:74]: dang

**Comment by katSchmid** [comment]
Gltf3DTileContent was part of our Model hack, no longer needed https://github.com/PropellerAero/cesium/commit/ab5e8839f7d510e8f23b30eee3fc7ec59e9984c3#diff-59026183589b26822ff7caa1c30b4a45c88ad1e17035f6ec5c67ec716e4c5179R4

File: package.json (hunk 3 above)
Comment (katSchmid) [reply package.json:75]: WE should Propeller hack this

File: gulpfile.js [hunk 5]
```diff
@@ -1476,7 +1477,8 @@ ${source}
     const assignmentName = basename(file, extname(file));
     if (publicModules.has(assignmentName)) {
       publicModules.delete(assignmentName);
-      source += `declare module "cesium/Source/${moduleId}" { import { ${assignmentName} } from 'cesium'; export default ${assignmentName}; }\n`;
+      //PROPELLER HACK
+      source += `declare module "cesium/Source/${moduleId}" { import { ${assignmentName} } from '@propelleraero/cesium'; export default ${assignmentName}; }\n`;
```
Comment (chris-cooper) [inline gulpfile.js]: It looks like the first `@propelleraero/` is missing here.

https://github.com/PropellerAero/cesium/blob/main/gulpfile.cjs#L1283

Comment (katSchmid) [reply gulpfile.js]: Not quite sure i understand correctly where we want to change?  moduleID?

Comment (chris-cooper) [reply gulpfile.js]:
```
source += `declare module "cesium/Source/${moduleId}" ...
vs
source += `declare module "@propelleraero/cesium/Source/${moduleId}" ...
```

**Comment by katSchmid** [comment]
Ok, lets change PR back to main?

**Comment by chris-cooper** [comment]
Looks good now! :tada:

Unfortunately a PR into main will mess up the history again.  Instead, you can now close this PR and force push this branch to main using a command like:

```
git push origin 1.99c:main --force
```

Note that while this resets main, the previous state of `main` will be preserved in the release tag e.g. https://github.com/PropellerAero/cesium/tree/propeller-1.96.2

**Pull request closed without merging**